File: code examples/nodejs/shopping_search.js
import fetch from 'node-fetch';

const username = 'YOUR_USERNAME';
const password = 'YOUR_PASSWORD';
const body = {
  'source': 'google_shopping_search',
  'query': 'adidas',
  'geo_location': 'New York,New York,United States',
  'pages': 2,
  'context': [
        {'key': 'sort_by', 'value': 'pd'},
        {'key': 'min_price', 'value': 20}
    ],
  'parse': true
};
const response = await fetch('https://realtime.oxylabs.io/v1/queries', {
  method: 'post',
  body: JSON.stringify(body),
  headers: {
    'Content-Type': 'application/json',
    'Authorization': 'Basic ' + Buffer.from(`${username}:${password}`).toString('base64'),
  }
});

console.log(await response.json());

File: code examples/python/shopping_search.py
import json
import requests

# Structure payload.
payload = {
    'source': 'google_shopping_search',
    'query': 'adidas',
    'geo_location': 'New York,New York,United States',
    'pages': 2,
    'context': [
        {'key': 'sort_by', 'value': 'pd'},
        {'key': 'min_price', 'value': 20},
    ],
    'parse': True
}

# Get response.
response = requests.request(
    'POST',
    'https://realtime.oxylabs.io/v1/queries',
    auth=('USERNAME', 'PASSWORD'),
    json=payload
)
print(response.json())

with open('shopping_search.json', 'w') as f:
    json.dump(response.json(), f, indent=2)